### DISCUSSION TITLE
How to use Homebrew version of perl?

### DISCUSSION THREAD

**Discussion by randomshinichi** [Homebrew/discussions#647] [Everyday usage]
https://github.com/mistydemeo/tigerbrew/pull/722/files

I updated a formula for Tigerbrew which needs perl 5.10+, but the OS comes with perl 5.8, and perl 5.22 is available as a formula, but apparently it is keg-only.

So is there a better way to specify that it should use Homebrew's perl to configure the Makefiles? I hardcoded `/usr/local/Cellar/perl/5.22.0/bin/` but I don't expect it to be accepted.

**Comment by carlocab** [comment]
Left you a comment: https://github.com/mistydemeo/tigerbrew/pull/722/files#r567164466

**Reply by randomshinichi** [reply]
Thanks! I ended up copying stuff out of real Homebrew's openssl.

**Comment by randomshinichi** [comment]
` ENV["PERL"] = Formula["perl"].opt_bin/"perl"`
Even though the Homebrew perl isn't actually in *opt/bin, somehow it still finds it (well technically it is in `/usr/local/opt/perl/bin`, guess it finds that automatically)

**Reply by carlocab** [reply]
That's because `Formula["perl"].opt_bin == "#{HOMEBREW_PREFIX}/opt/perl/bin"`.

You can see this with:
```
❯ brew ruby -e 'puts "perl".f.opt_bin'
/usr/local/opt/perl/bin
```
